mreg_cli/api/models.py

```diff
@@ -255,42 +263,134 @@ def is_delegated(self) -> bool:
         return True
 
 
-class Role(FrozenModelWithTimestamps, APIMixin["Role"]):
-    """Model for a role.
+class HostPolicy(FrozenModel):
+    """Base model for Host Policy objects.
+
+    Note:
+    ----
+    Host policy models in MREG have a different `created_at` field than
+    other models. It is called `create_date` and is a date - not a datetime.
+
+    This model has a custom validator to validate and convert the `create_date`
+    field to a datetime object with the expected `created_at` name.
 
-    Note that HostPolicy throws out `create_date` as the date only, not as a
-    proper datetime object, and not with the expected name `created_at`.
     """
 
-    id: int  # noqa: A003
-    created_at: datetime = Field(..., validation_alias=AliasChoices("create_date", "created_at"))
-    hosts: NameList
-    atoms: NameList
-    description: str
+    created_at_tz_naive: datetime = Field(
+        ...,
+        validation_alias=AliasChoices("create_date", "created_at", "created_at_tz_naive"),
+        exclude=True,
+        repr=False,
+    )
+    """Constructed datetime field from `create_date` in the API.
+
+    WARNING
+    ----
+    DO NOT USE THIS FIELD FOR TIMEZONE-AWARE COMPARISONS!
+    Always use `created_at` instead when comparing with timezone-aware
+    fields such as `update_time`."""
+    updated_at: datetime
     name: str
-    labels: list[int]
+    description: str
+
+    @field_validator("created_at_tz_naive", mode="before")
+    @classmethod
+    def validate_created_at(cls, value: Any) -> datetime:
+        """Convert a datetime string to a datetime object.
+
+        :param value: The input value - should be a datetime string.
+        :returns: The input value converted to a datetime object.
+        """
+        # Fast path for str (most likely input type)
+        if isinstance(value, str):
+            return datetime.fromisoformat(value)
+        elif isinstance(value, datetime):
+            return value
+        elif isinstance(value, date):
+            return datetime.combine(value, datetime.min.time())
+        return value  # let pydantic throw the ValidationError
+
+    @computed_field
+    def created_at(self) -> datetime:
+        """Creation time."""
+        return self.created_at_tz_naive.replace(tzinfo=self.updated_at.tzinfo)
 
-    @field_validator("created_at", mode="before")
     @classmethod
-    def validate_created_at(cls, value: str) -> datetime:
-        """Validate and convert the created_at field to datetime.
+    def get_by_name(cls, name: str) -> Atom | Role:
+        """Get an Atom or Role by name.
+
+        :param name: The name to search for.
+        :returns: The Atom or Role if found.
+        :raises CliWarning: If the Atom or Role is not found.
+        """
+        role_or_atom: Role | Atom | None = None
+        for func in [Atom.get_by_name, Role.get_by_name]:
```

Comment:
I'm not *completely* sold on this way of using exceptions (maybe we should have an option for getting these types that may return None), but the current implementation does work.

Reply:
Yes, that would be ideal. I am not overly happy catching `CliWarning`, but given the available methods it seemed to be the method involving the least boilerplate.

Reply:
Yeah, good point. We will have to do a second cleaning pass of this at some point anyway. There are more pressing issues to deal with.